File: library-appcompat/src/main/java/com/cz/android/sample/library/permission/PermissionViewModel.java
package com.cz.android.sample.library.permission;


import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;
import android.support.annotation.MainThread;

/**
 * @author Created by cz
 * @date 2020/7/28 5:02 PM
 * @email bingo110@126.com
 */
public class PermissionViewModel extends ViewModel {

    private final MutableLiveData<PermissionObserver> observerMap=new MutableLiveData<>();

    @MainThread
    PermissionObserver getObserver(){
        return observerMap.getValue();
    }

    @MainThread
    public void addObserver(PermissionObserver observer){
        observerMap.setValue(observer);
    }
}

File: extension/src/main/java/com/github/jackchen/android/sample/library/function/permission/PermissionViewModel.kt
package com.github.jackchen.android.sample.library.function.permission

import androidx.annotation.MainThread
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer

/**
 * @author Created by cz
 * @date 2020/7/28 5:02 PM
 * @email bingo110@126.com
 */
class PermissionViewModel : ViewModel() {
    private val permissionLiveData = MutableLiveData<PermissionResult>()

    fun setPermissionResult(permissionResult: PermissionResult) {
        permissionLiveData.value = permissionResult
    }

    @MainThread
    fun addObserver(lifecycleOwner: LifecycleOwner, observer: Observer<PermissionResult>) {
        permissionLiveData.observe(lifecycleOwner, observer)
    }
}